homeassistant/components/deconz/siren.py

```diff
@@ -0,0 +1,78 @@
+"""Support for deCONZ siren."""
+
+from pydeconz.light import Siren
+
+from homeassistant.components.siren import (
+    ATTR_DURATION,
+    DOMAIN,
+    SUPPORT_DURATION,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SirenEntity,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import NEW_LIGHT
+from .deconz_device import DeconzDevice
+from .gateway import get_gateway_from_config_entry
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up sirens for deCONZ component."""
+    gateway = get_gateway_from_config_entry(hass, config_entry)
+    gateway.entities[DOMAIN] = set()
+
+    @callback
+    def async_add_siren(lights=gateway.api.lights.values()):
+        """Add siren from deCONZ."""
+        entities = []
+
+        for light in lights:
+
+            if (
+                isinstance(light, Siren)
+                and light.unique_id not in gateway.entities[DOMAIN]
+            ):
+                entities.append(DeconzSiren(light, gateway))
+
+        if entities:
+            async_add_entities(entities)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(
+            hass, gateway.async_signal_new_device(NEW_LIGHT), async_add_siren
```

Comment:
Is this const correct?

Reply:
Yes. There are only a couple of major endpoints (lights/sensor)  which all different device types are exposed from. In this case siren is exposed as a lights resource